### ISSUE TITLE
helmfile not producing expected output after addition of strategicMergePatches

### ISSUE THREAD

**Issue by kyounger** [roboll/helmfile#1318]
I am attempting to use the new `strategicMergePatches` to insert an initContainer into a StatefulSet definition from an upstream chart. I'm running into an issue where it generates yaml for templates for subcharts that should be conditionally excluded by default. (The parent Chart.yaml defines the subchart with a `- condition: sidecarinjector.Enabled`, and that value is set to `false` in the default values.yaml.)

Here is my minimal helmfile that I'm using to reproduce the issue:

```
repositories:
- name: cloudbees
  url: https://charts.cloudbees.com/public/cloudbees
releases:
- name: cbcore
  namespace: cloudbees
  chart: cloudbees/cloudbees-core
  version: 3.15.1
  values:
  - OperationsCenter:
      Hostname: example.org
  strategicMergePatches:
  - apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      name: cjoc
    spec:
      template:
        spec:
          initContainers:
          - name: populate-plugins
            command:
            - sleep
            - "2"
```

If I run a `helmfile template` on that helmfile, then I see these resources defined in it, which indicates that they are being generated as part of what will get sent to the cluster if I were to run apply/sync.

```
cloudbees-sidecar-injector            ConfigMap          
cloudbees-sidecar-injector            Deployment         
cloudbees-sidecar-injector            PodDisruptionBudget
cloudbees-sidecar-injector            Service            
cloudbees-sidecar-injector            ServiceAccount     
cloudbees-sidecar-injector-init       Job                
cloudbees-sidecar-injector-init       Role               
cloudbees-sidecar-injector-init       RoleBinding        
cloudbees-sidecar-injector-init       ServiceAccount     
```

Additionally, I'm seeing that the comments in the resulting yaml refer to helmx, rather than the original template name. This would be minor, but triggers an unneeded sync when running `helmfile apply`, since it shows up in a `helmfile diff` if not previously using strategicMergePatches.

Snippet of the output yaml, note the `helmx.all.yaml`
```
    resources:
      - ingresses/status
    verbs:
      - update
---
# Source: cloudbees-core/templates/helmx.all.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app.kubernetes.io/instance: cbcore
```
Maybe a separate bug? Happy to split it off.

All that being said, the strategicMergePatch is actually updating the resource I want and inserting the yaml as expected, so that's working great! I just can't run the `helmfile apply` to get into the cluster! 😆

**Comment by kyounger** [comment]
Adjusted title of issue to reflect that this is not just for the diff function. I replicated it in a single helmfile (included above) and can reproduce with helm template. I did originally run into this because I added the strategicMergePatches into my release definition and had a lot more in my diff than expected.

**Comment by mumoshu** [comment]
@kyounger Hey! Thanks for reporting.

I just realized that the first one is duplicated of #1365 which has been already fixed since Helmfile v0.122.2. Would you mind confirming it works on you, too?

For the second issue, good catch! But I think there won't be an easy fix. Let me try a few days to see if I could find some fix for that.

**Comment by kyounger** [comment]
This was also fixed by #1363 I think.
